src/dmclock_server.h

```diff
@@ -1777,9 +1778,12 @@ namespace crimson {
 
 	    l.unlock();
 	    if (!this->finishing) {
-	      typename super::DataGuard g(this->data_mtx);
-	      schedule_request();
-	    }
+	      do {
+ 	        typename super::DataGuard g(this->data_mtx);
+ 	        if (schedule_request() == super::NextReqType::future)
+ 	          break;
+	      } while (this->request_count());
```

Comment:
nit,
```suggestion
	      } while (!empty());
```

Reply:
Good suggestion. this pointer is necessary for empty().